src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -102,7 +124,137 @@ class StreamReader {
 
 protected:
     explicit StreamReader() = default;
+
+    /**
+     * Templated implementation of `decode_range` that uses `log_event_to_string` to convert
+     * `log_event` to a string for the returned result.
+     *
+     * @tparam LogEvent
+     * @tparam ToStringFunc Function to convert a log event into a string.
+     * @param begin_idx
+     * @param end_idx
+     * @param filtered_log_event_map
+     * @param log_events
+     * @param use_filter
+     * @param log_event_to_string
+     * @return See `decode_range`.
```

Comment:
Sorry for the lack of context at https://github.com/y-scope/clp-ffi-js/pull/35#discussion_r1888811976 

```suggestion
     * @return See `decode_range`.
     * @throw Propagates `ToStringFunc`'s exception.
```